services/web/server/tests/unit/with_dbs/03/products/test_products_handlers.py

```diff
@@ -0,0 +1,101 @@
+# pylint: disable=redefined-outer-name
+# pylint: disable=unused-argument
+# pylint: disable=unused-variable
+# pylint: disable=too-many-arguments
+
+
+import operator
+from decimal import Decimal
+
+import pytest
+from aiohttp import web
+from aiohttp.test_utils import TestClient
+from models_library.api_schemas_webserver.product import GetProduct
+from models_library.products import ProductName
+from pytest_simcore.helpers.utils_assert import assert_status
+from pytest_simcore.helpers.utils_login import UserInfoDict
+from simcore_service_webserver._constants import X_PRODUCT_NAME_HEADER
+from simcore_service_webserver.db.models import UserRole
+
+
+@pytest.mark.parametrize(
+    "user_role,expected",
+    [
+        (UserRole.ANONYMOUS, web.HTTPUnauthorized),
+        (UserRole.GUEST, web.HTTPForbidden),
+        *((role, web.HTTPOk) for role in UserRole if role >= UserRole.USER),
+    ],
+)
+async def test_get_product_price_when_undefined(
+    client: TestClient,
+    logged_user: UserInfoDict,
+    expected: type[web.HTTPException],
+    new_osparc_price: Decimal,
+):
+    response = await client.get("/v0/credits-price")
+    data, error = await assert_status(response, expected)
+
+    if not error:
+        assert data["usdPerCredit"] == new_osparc_price
+
+
+@pytest.mark.parametrize(
+    "user_role,expected",
+    [
+        (UserRole.ANONYMOUS, web.HTTPUnauthorized),
+        (UserRole.GUEST, web.HTTPForbidden),
+        (UserRole.USER, web.HTTPForbidden),
+        (UserRole.TESTER, web.HTTPForbidden),
+        (UserRole.PRODUCT_OWNER, web.HTTPOk),
+        (UserRole.ADMIN, web.HTTPForbidden),
+    ],
+)
+async def test_get_product_access_rights(
+    client: TestClient,
+    logged_user: UserInfoDict,
+    expected: type[web.HTTPException],
+    new_osparc_price: Decimal,
+):
+    response = await client.get("/v0/products/current")
+    data, error = await assert_status(response, expected)
+    assert operator.xor(data is None, error is None)
```

Comment:
I have a feeling this is always the case and this check does not bring much.
Maybe checking that data is not None in case of HTTPOk is better?

Reply:
why do you have this feeling? :-) 
This makes sure that only one is `None` and not both. The previous check, in line 60 is checking what you are suggesting

Reply:
missed that one